src/main/java/org/fungover/mmotodo/service/AuthService.java

```diff
@@ -0,0 +1,151 @@
+package org.fungover.mmotodo.service;
+
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.fungover.mmotodo.dto.GithubUser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+
+
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+
+import org.springframework.stereotype.Service;
+
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+
+import java.util.Base64;
+import java.util.Map;
+
+@Service
+public class AuthService {
+
+    @Value("${spring.security.oauth2.client.registration.github.clientId}")
+    private String clientId;
+
+    @Value("${spring.security.oauth2.client.registration.github.clientSecret}")
+    private String clientSecret;
+
+    @Value("${github.api.logout.url}")
+    private String githubDeleteUrl;
+
+    private final OAuth2AuthorizedClientService authorizedClientService;
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthService.class);
+
+    public AuthService(OAuth2AuthorizedClientService authorizedClientService) {
+        this.authorizedClientService = authorizedClientService;
+    }
+
+
+    public GithubUser getUserData(@AuthenticationPrincipal OAuth2User principal) {
+        Map<String, Object> attributes = principal.getAttributes();
+        String userName = attributes.get("login").toString();
+        String name = attributes.get("name").toString();
+        int id = (int) attributes.get("id");
+        String avatarUrl = attributes.get("avatar_url").toString();
+        String githubProfileUrl = attributes.get("html_url").toString();
+        String email = attributes.get("email") != null ? attributes.get("email").toString() : null;
+        String createdAt = attributes.get("created_at").toString();
+        String updatedAt = attributes.get("updated_at").toString();
+
+
+        return new GithubUser(userName, name, id, avatarUrl, githubProfileUrl, email, createdAt, updatedAt);
+
+    }
+
```

Comment:
Is email the only null check needed here? What about if the id can't be cast to an int?